chain/badtscache.go

```diff
@@ -43,7 +43,7 @@ func (bbr BadBlockReason) String() string {
 }
 
 func NewBadBlockCache() *BadBlockCache {
-	cache, err := lru.NewARC[cid.Cid, BadBlockReason](build.BadBlockCacheSize)
+	cache, err := arc.NewARC[cid.Cid, BadBlockReason](build.BadBlockCacheSize)
```

Comment:
We don't really need this for the libp2p upgrade, right ? Do you think we can do this in a separate PR ?

Reply:
Yes you need this, due to a breaking change in that library introduced in a patch release (see https://github.com/hashicorp/golang-lru/issues/142). I agree that this cherry-pick should be left as is.